File: library/src/test/java/org/crispy_fish/filetype/staging/StagingFilenameFilterTest.java
package org.crispy_fish.filetype.staging;

import java.util.regex.Pattern;

import org.coner.crispy_fish.filetype.staging.StagingFilenameFilter;
import org.coner.crispy_fish.filetype.staging.StagingFilenames;
import org.junit.*;
import org.junit.rules.TemporaryFolder;
import org.junit.runner.RunWith;
import org.mockito.runners.MockitoJUnitRunner;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(MockitoJUnitRunner.class)
public class StagingFilenameFilterTest {

    private StagingFilenameFilter stagingFilenameFilter;

    private static final String EVENT_CONTROL_FILE_ORIGINAL_STAGING_BASE_NAME = "bar";
    private static final Pattern STAGING_ORIGINAL_FILE_PATTERN = StagingFilenames.INSTANCE.getORIGINAL_FILE_DAY_1();

    @Rule
    public final TemporaryFolder dir = new TemporaryFolder();

    @Before
    public void setup() {
        stagingFilenameFilter = new StagingFilenameFilter(
                EVENT_CONTROL_FILE_ORIGINAL_STAGING_BASE_NAME,
                STAGING_ORIGINAL_FILE_PATTERN
        );
    }

    @Test
    public void whenExtensionIsNormalStagingItShouldAccept() {
        String name = "bar.st1";

        boolean actual = stagingFilenameFilter.accept(dir.getRoot(), name);

        assertThat(actual).isTrue();
    }

    @Test
    public void whenExtensionIsAbsentItShouldReject() {
        String name = "bar";

        boolean actual = stagingFilenameFilter.accept(dir.getRoot(), name);

        assertThat(actual).isFalse();
    }

    @Test
    public void whenExtensionIsPresentButWrongItShouldReject() {
        String name = "bar.txt";

        boolean actual = stagingFilenameFilter.accept(dir.getRoot(), name);

        assertThat(actual).isFalse();
    }

    @Test
    public void whenExtensionIsStagingLogItShouldReject() {
        String name = "bar.st1_log";

        boolean actual = stagingFilenameFilter.accept(dir.getRoot(), name);

        assertThat(actual).isFalse();
    }
}

File: library/src/test/kotlin/org/coner/crispyfish/filetype/staging/StagingFilenameFilterTest.kt
package org.coner.crispyfish.filetype.staging

import org.junit.*
import org.junit.rules.TemporaryFolder
import org.junit.runner.RunWith

import org.assertj.core.api.Assertions.assertThat
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class StagingFilenameFilterTest {

    private lateinit var stagingFilenameFilter: StagingFilenameFilter

    @Rule @JvmField
    val dir = TemporaryFolder()

    @Before
    fun setup() {
        stagingFilenameFilter = StagingFilenameFilter(
                EVENT_CONTROL_FILE_ORIGINAL_STAGING_BASE_NAME,
                STAGING_ORIGINAL_FILE_PATTERN
        )
    }

    @Test
    fun whenExtensionIsNormalStagingItShouldAccept() {
        val name = "bar.st1"

        val actual = stagingFilenameFilter.accept(dir.root, name)

        assertThat(actual).isTrue()
    }

    @Test
    fun whenExtensionIsAbsentItShouldReject() {
        val name = "bar"

        val actual = stagingFilenameFilter.accept(dir.root, name)

        assertThat(actual).isFalse()
    }

    @Test
    fun whenExtensionIsPresentButWrongItShouldReject() {
        val name = "bar.txt"

        val actual = stagingFilenameFilter.accept(dir.root, name)

        assertThat(actual).isFalse()
    }

    @Test
    fun whenExtensionIsStagingLogItShouldReject() {
        val name = "bar.st1_log"

        val actual = stagingFilenameFilter.accept(dir.root, name)

        assertThat(actual).isFalse()
    }

    companion object {

        private val EVENT_CONTROL_FILE_ORIGINAL_STAGING_BASE_NAME = "bar"
        private val STAGING_ORIGINAL_FILE_PATTERN = StagingFilenames.ORIGINAL_FILE_DAY_1
    }
}
